source/common/config/grpc_mux_impl.cc

```diff
@@ -138,13 +147,12 @@ void GrpcMuxImpl::onDiscoveryResponse(
       // xDS server sends an empty list of ClusterLoadAssignment resources. we'll accept
       // this update. no need to send a discovery request, as we don't watch for anything.
       api_state_[type_url].request_.set_version_info(message->version_info());
+      return;
     } else {
-      // No watches and we have resources - this should not happen. send a NACK (by not
-      // updating the version).
       ENVOY_LOG(warn, "Ignoring unwatched type URL {}", type_url);
       queueDiscoveryRequest(type_url);
+      return;
     }
-    return;
```

Comment:
Why this change?

Reply:
That is unintentional, I will clean it up.